src/frontend/src/components/forms/ApiForm.tsx

```diff
@@ -390,16 +410,18 @@ export function ApiForm({ id, props }: { id: string; props: ApiFormProps }) {
             {props.preFormWarning}
           </Alert>
         )}
-        <Stack spacing="xs">
-          {Object.entries(props.fields ?? {}).map(([fieldName, field]) => (
-            <ApiFormField
-              key={fieldName}
-              fieldName={fieldName}
-              definition={field}
-              control={form.control}
-            />
-          ))}
-        </Stack>
+        <FormProvider {...form}>
```

Comment:
Why do we need that form provider here? The form ref gets already passed to the fields via the control ref.

Reply:
As far as I could tell, the control ref is only for a specific field, not the whole form? I need the whole form dataset to pass through to `adjustFilters` for related fields (see below)